pyanaconda/modules/storage/bootloader/base.py

```diff
@@ -116,24 +117,31 @@ def partition_p(arg):
         for nic in config:
             ip_params.add("%s:%s" % (nic, ",".join(sorted(config[nic]))))
 
-        # update the set
-        self.clear()
-        self.update(rest)
-        self.update(ip_params)
-        return self
+        # update the list
+        self._arguments.clear()
+        self._arguments.extend(rest)
+        self._arguments.extend(ip_params)
+        return self._arguments
 
     def __str__(self):
+        """Convert the argument list to string.
+
+        Merges ip= arguments targeting the same interface as a side effect.
+        """
         self._merge_ip()
-        return " ".join(list(self))
+        return " ".join(list(self._arguments))
 
-    def add(self, key):
-        self.discard(key)
-        super().add(key)
+    def __iter__(self):
+        return iter(self._arguments)
 
     def update(self, sequence):
+        """Add arguments from some other iterable object.
+
+        :param sequence: iterable object
+        """
         for key in sequence:
-            self.discard(key)
-            self.add(key)
```

Comment:
I think that the effect of `discard` and `add` was that the argument was always appended at the end. That's no longer true.

Reply:
Btw. the `update` method should just call `self.add` on all items of the sequence and only the `add` method should implement the logic for adding an item into the list.

Reply:
Sorry, fixed.